extensions/wikia/AdEngine/js/provider/gpt/googleTag.js

```diff
@@ -89,71 +89,118 @@ define('ext.wikia.adEngine.provider.gpt.googleTag', [
 			}
 		});
-	};
-
-	GoogleTag.prototype.push = function (callback) {
-		window.googletag.cmd.push(callback);
-	};
+	}
 
-	GoogleTag.prototype.flush = function () {
-		if (!this.isInitialized()) {
+	function flush() {
+		if (!isInitialized()) {
 			log(['flush', 'done', 'No slots to flush'], 'info', logGroup);
 			return;
 		}
 
-		this.push(function () {
+		push(function () {
 			log(['flush', 'start'], 'info', logGroup);
 
 			log(['flush', 'refresh', slotQueue], 'debug', logGroup);
 			if (slotQueue.length) {
-				pubAds.refresh(slotQueue, {changeCorrelator: false});
+				window.googletag.pubads().refresh(slotQueue, {changeCorrelator: false});
 				slotQueue = [];
 			}
 
 			log(['flush', 'done'], 'info', logGroup);
 		});
-	};
+	}
 
-	GoogleTag.prototype.addSlot = function (adElement) {
+	function addSlot(adElement) {
 		var sizes = adElement.getSizes(),
-			slot = slots[adElement.getId()];
+			slotId = adElement.getId(),
+			slot = googleSlots.getSlot(slotId);
 
 		log(['addSlot', adElement], 'debug', logGroup);
 
 		adElement.setPageLevelParams(pageLevelParams);
 		if (!slot) {
 			if (sizes) {
-				slot = window.googletag.defineSlot(adElement.getSlotPath(), sizes, adElement.getId());
+				slot = window.googletag.defineSlot(adElement.getSlotPath(), sizes, slotId);
 			} else {
-				slot = window.googletag.defineOutOfPageSlot(adElement.getSlotPath(), adElement.getId());
+				slot = window.googletag.defineOutOfPageSlot(adElement.getSlotPath(), slotId);
 			}
-			slot.addService(pubAds);
-			window.googletag.display(adElement.getId());
-			slots[adElement.getId()] = slot;
+			slot.addService(window.googletag.pubads());
+			window.googletag.display(slotId);
+			googleSlots.addSlot(slot);
 		}
 
 		adElement.configureSlot(slot);
 		slotQueue.push(slot);
 
 		return slot;
-	};
+	}
 
-	GoogleTag.prototype.registerCallback = function (id, callback) {
+	function registerCallback(id, callback) {
 		log(['registerCallback', id], 'info', logGroup);
 		registeredCallbacks[id] = callback;
-	};
+	}
 
-	GoogleTag.prototype.onAdLoad = function (slotName, element, gptEvent, onAdLoadCallback) {
+	function onAdLoad(slotName, element, gptEvent, onAdLoadCallback) {
 		log(['onAdLoad', slotName], 'info', logGroup);
 		var iframe = element.getNode().querySelector('div[id*="_container_"] iframe');
 
 		onAdLoadCallback(element.getId(), gptEvent, iframe);
-	};
+	}
+
+	function destroySlots(slotsNames) {
+		var slotsToDestroy = [],
+			allSlots = window.googletag.getSlots(),
+			success;
+
+		// when nothing passed - destroy all slots
+		if (!slotsNames) {
+			slotsToDestroy = allSlots;
+		} else {
+			allSlots.forEach(function (slot) {
+				// google returns array
+				// - in our case it has always one element and this element is the one we are interested in
+				if (slotsNames.indexOf(slot.getTargeting('pos')[0]) > -1) {
+					slotsToDestroy.push(slot);
+				}
+			});
+		}
+
+		// this one protects us from removing all slots if slotsToDestroy is an empty array
```

Comment:
What do you think of removing this comment and replacing it with `else` logging to console that the array was empty and no slots were destroyed?